File: MyCustomersApp/Models/Customer.cs
namespace MyCustomersApp.Models;

public class Customer
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Number { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime UpdatedDate { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    
    public string FullName => !string.IsNullOrEmpty(FirstName) && !string.IsNullOrEmpty(LastName) 
        ? $"{FirstName} {LastName}" 
        : Name;
}

File: MyCustomersRN/src/types/Customer.ts
export interface Customer {
  id: string;
  name: string;
  email: string;
  number?: string;
  notes?: string;
  createdDate: string;
  updatedDate: string;
  firstName?: string;
  lastName?: string;
}

export interface CreateUpdateCustomerDto {
  name: string;
  email: string;
  number?: string;
  notes?: string;
  createdDate: string;
  updatedDate: string;
  firstName?: string;
  lastName?: string;
}

export interface PagedResponse<T> {
  items: T[];
  totalCount: number;
  pageSize: number;
  pageNumber: number;
  totalPages: number;
}